lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1337,6 +1341,23 @@ class HSADevice final : public KalmarDevice
         }
 
         return checksum.str();
+#else
+        // FNV-1a hashing, 64-bit version
+        const uint64_t FNV_prime = 0x100000001b3;
+        const uint64_t FNV_basis = 0xcbf29ce484222325;
+        uint64_t hash = FNV_basis;
+
+        const char *str = static_cast<const char *>(source);
+
+        // 104 is the proper size from Jack's research
```

Comment:
On a second thought, could you make it a macro? The rationale is because a new code object format is being developed so this value may have to be changed soon.